### ISSUE TITLE
Implement TextRank using a dfm 

### ISSUE THREAD

**Issue by brousseauj** [quanteda/quanteda#1722]
It would be great to be able to have a list of documents and quickly be able to summarize the relevant documents. Sometimes things like topfeatures() doesn't give enough context.

**Comment by koheiw** [comment]
Does "relevant" mean similarity? If so, it is coming soon.

**Comment by brousseauj** [comment]
Yeah I guess something that would like look at top features and then pull up documents that contain the most of the top features would be considered most "relevant".

**Comment by kbenoit** [comment]
TextRank is an implementation of PageRank to score sentences, in a way that the sentences with the highest scores can be considered good summaries of a document or a collection of documents. The methodology is described here: 
https://web.eecs.umich.edu/~mihalcea/papers/mihalcea.emnlp04.pdf.

@jwijffels has written an [R implementation](https://github.com/bnosac/textrank). Some approaches score sentences based on GloVe (embedding) scores but that package uses standard distance metrics from frequencies. Would be interesting to write a **quanteda** function based on this.

**Comment by jwijffels** [comment]
@brousseauj  The R implementation I've written in https://github.com/bnosac/textrank looks for word overlap. Nothing stops you from calculating another similarity metric (e.g. tfidf / embedding similarity / whichever other similarity metric). 
In fact, that's what I do in quite some projects, I now tend to use package ruimtehol https://cran.r-project.org/web/packages/ruimtehol/index.html to calculate sentence embeddings and feed them to textrank to summarise sentences

**Comment by brousseauj** [comment]
Thats really interesting. I'll have to check out that embedding package! Compared to other embedding packages like doc2vec or word2vec, how is this one? My knowledge of embeddings is quite limited so I apologize for the basic questions!

**Comment by jwijffels** [comment]
Ruimtehol is an r wrapper around starspace which allows to embed all the things: articles, sentences, words, bigrams, labels, tags, entities and entity relations. Anything quoi.
